src/Application/Commands/PatchFunds8Oct.php

```diff
@@ -0,0 +1,129 @@
+            ['id' => '9732','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000BlbDtYAJ'],
+            ['id' => '9735','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000BlbDtYAJ'],
+            ['id' => '9753','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000CqTzuYAF'],
+            ['id' => '9763','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000BlbDtYAJ'],
+            ['id' => '9767','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000CqOyeYAF'],
+            ['id' => '9768','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS000009MZDGYA4'],
+            ['id' => '9777','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000CqOyeYAF'],
+            ['id' => '9778','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000CqTzuYAF'],
+            ['id' => '9779','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000CqOyeYAF'],
+            ['id' => '9786','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000BlbDtYAJ'],
+            ['id' => '9806','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000CqOyeYAF'],
+            ['id' => '9814','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000CqTzuYAF'],
+            ['id' => '9824','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000CqTzuYAF'],
+            ['id' => '9838','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000CqTzuYAF'],
+            ['id' => '9842','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000CqTzuYAF'],
+            ['id' => '29091','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000BlbDtYAJ'],
+            ['id' => '29656','COUNT(DISTINCT cf.id)' => '2','GROUP_CONCAT(Fund.salesforceId)' => 'a09WS00000BDhATYA1,a09WS00000BlbDtYAJ']
+        ];
+        $campaignIds = array_column($Campaign, 'id');
+
+        /** @var Campaign[] $campaigns */
+        $campaigns = $this->campaignRepository->findBy(['id' => $campaignIds]);
+
+        return $campaigns;
+    }
+
+    private function switchMatchingToCorrectFunding(Donation $donation): void
+    {
+        $originallyMatched = $donation->getFundingWithdrawalTotal();
+        /** @psalm-suppress InternalMethod */
+        $this->allocator->releaseMatchFunds($donation);
+        $nowMatched = $this->allocator->allocateMatchFunds(donation: $donation, forceNotBigGive: true);
```

Comment:
should these two be wrapped in a transaction?

Reply:
I think because Redis is the source of truth there isn't a very easy way to do that. Ideally yes, but as it's small fractions of a second I think the risk is low compared to the increased risk of fund over allocation by delaying.